indicators/NO_BFLY_001_002/R/NO_BFLY_001_002.qmd

```diff
@@ -0,0 +1,873 @@
+
+-->
+```
+
+Ecosystem assets: Mid-, central-, east-, and southern Norway. Note that southern Norway only includes the old counties of Vestfold and Østfold, and the south region contain Rogaland and Vest Agder, but not Øst Agder. The mid or central region includes Trøndelag and Møre og Romsdal.
+
+Ecosystem accounting area: Norway (although N-Norway and several sub-regions are missing).
+
+Data needs to be aggregated to EA level before interpretation. In addition, only locations with complete sampling (three replicated per season) should be included to avoids seasonal biases.
+
+## 9. Analyses
+
+```{=html}
+<!--# 
+
+Use this header for documenting the analyses. Put code in seperate code chunks, and annotate the code in between using normal text (i.e. between the chunks, and try to avoid too many hashed out comments inside the code chunks). Add subheaders as needed. 
+
+Code folding is activated, meaning the code will be hidden by default in the html (one can click to expand it).
+
+Caching is also activated (from the top YAML), meaning that rendering to html will be quicker the second time you do it. This will create a folder inside you project folder (called INDICATORID_cache). Sometimes caching created problems because some operations are not rerun when they should be rerun. Try deleting the cash folder and try again.
+
+-->
+```
+
+These calculations with bootstrap sampling are implemented in the R-package [bombLepiSurv](https://github.com/NINAnor/bombLepiSurv).
+
+Install package like this.
+
+```{r}
+#| eval: false
+#| code-fold: show
+remotes::install_github("NINAnor/bombLepiSurv")
+library(bombLepiSurv)
+```
+
+Here I illustrate how to connect to the database. Since the database is only accessible from NINA, I will not run this part of the analyses, but instead cache the output files so that others can reproduce the general workflow, even if they cannot access the database.
+
+```{r connect}
+#| eval: false
+# get secret username and password. I just keep mine in a local .R file that I also put in the gitignore file.
+source(here::here("_pw.R"))
+# connect
+bombLepiSurv::humlesommerfConnect(
+  username = BOMB_USER,
+  password = BOMB_PASSWORD,
+  host = "ninradardata01.nina.no"
+)
+```
+
+Download data from 2023 as an example. 2024 data is not yet published (although it is made availble in a developement branch on GitHub).
+
+```{r}
+#| eval: false
+allLepiGressmark2023 <- 
+  getAllData(type = "butterflies", 
+habitat = "gressmark",
+    year = 2023,
+    language = "norsk") |>
+      mutate_if(bit64::is.integer64, as.numeric)
+    
+fig_allLepiGressmark2023 <- surveyBarplot(allLepiGressmark2023)
+saveRDS(fig_allLepiGressmark2023, here::here("indicators/NO_BFLY_001_002/img/fig_allLepiGressmark2023.rds"))
+```
+
+```{r fig-allLepiGressmark2023}
+#| fig-cap: 'Butterflies in grasslands 2023.'
+readRDS(here::here("indicators/NO_BFLY_001_002/img/fig_allLepiGressmark2023.rds"))
```

Comment:
Get error when running this line.
Probably problem with ggplot version.
Error in `geom_bar()`:
! Problem while computing aesthetics.
ℹ Error occurred in the 1st layer.
Caused by error in `scales_add_defaults()`:
! could not find function "scales_add_defaults"

Reply:
saved as png instead